src/completion-listener.ts

```diff
@@ -86,7 +86,8 @@ export class CompletionListener {
      * @returns A Promise that resolves if the success condition is met, or rejects otherwise.
      */
     listen(commands: Command[]): Promise<CloseEvent[]> {
-        const closeStreams = commands.map((command) => command.close);
+        const closeStreams = commands.map((command) => command.close.pipe(take(1)));
+
         return Rx.lastValueFrom(
             Rx.merge(...closeStreams).pipe(
                 bufferCount(closeStreams.length),
```

Comment:
> So how would u recommend proceeding?

I think that replacing these lines with `Rx.combineLatest()` + removing the `take(1)` above can achieve retaining the last emitted close event.

We can then also fix your edge case from https://github.com/open-cli-tools/concurrently/pull/464#issuecomment-1921230207 with a `filter()` operator that checks if every command is indeed closed.
This means waiting until every command emits another close event.

WDYT?

Reply:
Yea that sounds legit, at-least the first part, seems like `combineLatest` can do the job 👍  Ill try to change the implementation accordingly

Reply:
@gustavohenke I've pushed the changes, the only drawback using the `combineLatest` is that it actually keep the close events according to the registered stream **and not by the actual order they were emitted**, So beside accepting your offer I also added a `map` to stort it via the `startingDate` of the event, would love to hear if you got any better suggestion here 👍